### PR TITLE
feat: add TOON output format for LLM token optimization

### LINKED ISSUE #86: Add TOON output format for LLM token optimization
## Summary

Add TOON (Token-Oriented Object Notation) output format to reduce LLM token consumption by ~58.9% compared to JSON for list commands.

## Context

- **Current state**: Commands support `--json` flag (ADR 0003)
- **Problem**: JSON is token-expensive for LLMs (repeated keys, excessive punctuation)
- **Solution**: TOON library provides 40-60% token reduction
- **Use case**: AI plugin consumes CLI output for triage/review/fix workflows

## Scope

**High-value commands** (Phase 1):
1. `issue sub-issue list`
2. `issue dependency list`
3. `pr review thread list`
4. `plugin list`
5. `plugin search`
6. `issue type list`

## Implementation Tasks

- [ ] Install `@byjohann/toon` dependency
- [ ] Create `src/lib/toon-output.ts` utility module
- [ ] Extend `src/lib/json-output.ts` with format selection
- [ ] Add `--format <json|toon>` flag to 6 commands
- [ ] Add unit tests for TOON encoding
- [ ] Update integration tests for format flag
- [ ] Document TOON usage in CLAUDE.md
- [ ] Create ADR 0006 (✅ completed)
- [ ] Benchmark actual token savings

## Acceptance Criteria

- ✅ TOON output produces valid, parseable format
- ✅ Token count reduced by 50%+ vs JSON on sample data
- ✅ All tests pass with new flag
- ✅ Backward compatible (existing --json unchanged)
- ✅ Documentation explains TOON benefits

## Technical Details

**Default delimiter**: Tab (`\t`) for best tokenization (58.9% savings)

**Example output**:
\`\`\`
# JSON (257 tokens)
[{"number":124,"title":"Test","state":"OPEN"}]

# TOON (105 tokens, 58.9% reduction)
[1	]{number	title	state}:
  124	Test	OPEN
\`\`\`

## References

- TOON Repository: https://github.com/johannschopplich/toon
- ADR 0006: docs-dev/adr/0006-toon-output-format.md
- Related: ADR 0003 (JSON Output Implementation)

## Estimated Effort

- Development: 2-3 hours
- Testing: 1 hour
- Documentation: 1 hour
- **Total: 4-5 hours**

### PR DESCRIPTION
**Pull request by amondnet**
## Summary

Adds TOON (Token-Oriented Object Notation) output format to reduce LLM token consumption by 58.9% compared to JSON when processing CLI output. The implementation provides a `--format toon` flag for 6 high-value list commands while maintaining full backward compatibility with JSON as the default.

### Core Components

- **TOON Library Integration**: Added `@byjohann/toon@0.4.1` dependency
- **Output Utilities** (`src/lib/toon-output.ts`):
  - `encodeToon()` - Encode data to TOON format with tab delimiters
  - `outputToon()` - Output TOON to stdout
- **Unified Output Interface** (`src/lib/json-output.ts`):
  - `outputData()` - Supports both JSON and TOON formats
  - `isValidFormat()` / `validateFormat()` - Format validation
  - Maintains existing `outputJson()` and field filtering

### Commands Updated

All 6 commands now support `--format json|toon` flag:

1. `gh please issue sub-issue list <issue>`
2. `gh please issue dependency list <issue>`
3. `gh please issue type list`
4. `gh please pr review thread list <pr>`
5. `gh please plugin list`
6. `gh please plugin search [query]`

### Token Savings

- **58.9% reduction** with tab delimiters (vs JSON)
- **49.1% reduction** with comma delimiters
- Tab delimiters chosen for optimal tokenization

Example:
```bash
# JSON output (default) - 245 tokens
gh please issue sub-issue list 123 --json

# TOON output - 101 tokens (58.8% reduction)
gh please issue sub-issue list 123 --format toon
```

### Test Coverage

- **80 total tests**, all passing:
  - TOON encoding: 14 tests
  - JSON output utilities: 31 tests
  - Format validation: 15 tests
  - Command integration: 10 tests
  - Validation integration: 25 tests (including 15 unit tests)

### Test Files

- `test/lib/toon-output.test.ts` - Core TOON encoding
- `test/lib/json-output.test.ts` - Output utilities
- `test/lib/format-validation.test.ts` - Format validation
- `test/commands/format-flag-integration.test.ts` - Command wiring
- `test/commands/format-validation-integration.test.ts` - Invalid format handling

### Critical Issues Fixed

1. **✅ Reverted unstaged changes** to `src/lib/github-graphql.ts`
   - Restored detailed error messages with troubleshooting guidance

2. **✅ Added command integration tests** (`test/commands/format-flag-integration.test.ts`)
   - 10 comprehensive tests covering TOON/JSON output, field filtering, edge cases
   - Tests verify command-level wiring of --format flag

3. **✅ Added format value validation**
   - Runtime validation with clear error messages
   - Validation in all 6 commands + `outputData()` function
   - 40 tests covering validation (15 unit + 25 integration)
   - Error format: `Invalid output format: xml. Supported formats: json, toon`

## Documentation

- **ADR 0006**: Architecture Decision Record documenting rationale, benchmarks, alternatives
- **GitHub Issue #86**: Implementation tasks and acceptance criteria

## Examples

```bash
# JSON output (default)
gh please issue sub-issue list 123 --json
[
  {
    "number": 124,
    "title": "Implement auth",
    "state": "OPEN"
  }
]

# TOON output (58.9% token reduction)
gh please issue sub-issue list 123 --format toon
[1	]{number	title	state}:
  124	Implement auth	OPEN

# Field filtering works with both formats
gh please plugin list --format toon --json name,version
[2	]{name	version}:
  ai	0.3.0
  speckit	0.1.0

# Invalid format - clear error message
gh please issue sub-issue list 123 --format xml
Error: Invalid output format: xml. Supported formats: json, toon
```

## Backward Compatibility

- ✅ JSON remains the default format
- ✅ TOON is opt-in via `--format toon` flag
- ✅ All existing commands work unchanged
- ✅ Field filtering works with both formats
- ✅ Progress messages suppressed in structured output mode

## Related

- Closes #86
- ADR: `docs-dev/adr/0006-toon-output-format.md`
- TOON Library: https://github.com/johannschopplich/toon
- Token Benchmarks: See ADR for detailed analysis

## Files Changed

- 22 files changed, 1817 insertions(+), 85 deletions(-)
- 7 new source files (TOON utilities, validation)
- 5 test files (80 tests total)
- 6 commands updated with --format flag
- 1 ADR documenting decision

### CODE REVIEW

File: src/commands/issue/dependency.ts [hunk 1]
```diff
@@ -112,7 +113,8 @@ export function createDependencyCommand(): Command {
     .argument('<issue>', 'Issue number')
     .option('-R, --repo <owner/repo>', 'Repository in owner/repo format')
     .option('--json [fields]', 'Output in JSON format with optional field selection (number,title,state,nodeId,url)')
-    .action(async (issueStr: string, options: { repo?: string, json?: string | boolean }) => {
+    .option('--format <format>', 'Output format: json or toon (default: json)', 'json')
```
[the remark this answers is not in the document]
Comment (amondnet) [reply src/commands/issue/dependency.ts]: ✅ Applied in [commit 04fc06d](https://github.com/pleaseai/gh-please/commit/04fc06d)
- File: `src/commands/issue/dependency.ts`
- Change: Removed default value from --format option

Thanks for catching this bug, @gemini-code-assist!

File: src/commands/pr/review/thread-list.ts [hunk 2]
```diff
@@ -17,10 +18,11 @@ export function createThreadListCommand(): Command {
     .option('--unresolved-only', 'Show only unresolved threads')
     .option('-R, --repo <owner/repo>', 'Repository in owner/repo format')
     .option('--json [fields]', 'Output in JSON format with optional field selection (nodeId,isResolved,path,line,resolvedBy,firstCommentBody,url)')
+    .option('--format <format>', 'Output format: json or toon (default: json)', 'json')
```
[the remark this answers is not in the document]
Comment (amondnet) [reply src/commands/pr/review/thread-list.ts]: ✅ Applied in [commit 04fc06d](https://github.com/pleaseai/gh-please/commit/04fc06d)
- File: `src/commands/pr/review/thread-list.ts`
- Change: Removed default value from --format option

Thanks for catching this bug, @gemini-code-assist!
